### PR TITLE
CNS-799: uint conversions and static analysis tools fixes

### PR DESCRIPTION
**Pull request by oren-lava**
The static analysis commands used:

`golangci-lint run --disable-all --enable gosec,govet,staticcheck`
`golangci-lint run --disable-all --presets bugs ./x/... -v`

### CODE REVIEW

File: x/conflict/keeper/vote.go [hunk 1]
```diff
@@ -122,7 +122,11 @@ func (k Keeper) HandleAndCloseVote(ctx sdk.Context, conflictVote types.ConflictV
 			providersWithoutVote = append(providersWithoutVote, vote.Address)
 			bail := stake
 			bail.Quo(sdk.NewIntFromUint64(BailStakeDiv))
-			k.pairingKeeper.JailEntry(ctx, accAddress, conflictVote.ChainID, conflictVote.VoteStartBlock, blocksToSave, sdk.NewCoin(k.stakingKeeper.BondDenom(ctx), bail))
+			err = k.pairingKeeper.JailEntry(ctx, accAddress, conflictVote.ChainID, conflictVote.VoteStartBlock, blocksToSave, sdk.NewCoin(k.stakingKeeper.BondDenom(ctx), bail))
+			if err != nil {
+				utils.LavaFormatWarning("jailing failed at vote conflict", err)
+				// not skiiping to continue to slash
```
Comment (shleikes) [inline x/conflict/keeper/vote.go:128]:
```suggestion
				// not skipping to continue to slash
```

File: x/dualstaking/keeper/helpers_test.go [hunk 2]
```diff
@@ -41,15 +41,18 @@ func (ts *tester) setupForDelegation(delegatorCount, stakedCount, unstakedCount,
 
 	ts.addClients(delegatorCount)
 
-	ts.addProviders(stakedCount)
+	err := ts.addProviders(stakedCount)
+	require.Nil(ts.T, err)
```
Comment (shleikes) [inline x/dualstaking/keeper/helpers_test.go:45]:
```suggestion
	require.NoError(ts.T, err)
```

File: x/dualstaking/keeper/helpers_test.go [hunk 3]
```diff
@@ -41,15 +41,18 @@ func (ts *tester) setupForDelegation(delegatorCount, stakedCount, unstakedCount,
...
 		err := ts.StakeProvider(addr, ts.spec, testStake)
 		require.NoError(ts.T, err)
 	}
 
-	ts.addProviders(unstakedCount)
-	ts.addProviders(unstakingCount)
+	err = ts.addProviders(unstakedCount)
+	require.Nil(ts.T, err)
+	err = ts.addProviders(unstakingCount)
+	require.Nil(ts.T, err)
```
Comment (shleikes) [inline x/dualstaking/keeper/helpers_test.go:55]:
```suggestion
	err = ts.addProviders(unstakedCount)
	require.NoError(ts.T, err)
	err = ts.addProviders(unstakingCount)
	require.NoError(ts.T, err)
```

File: x/dualstaking/keeper/hooks.go [hunk 4]
```diff
@@ -152,7 +152,10 @@ func (h Hooks) BeforeDelegationRemoved(ctx sdk.Context, delAddr sdk.AccAddress,
 	}
 	validator, err := h.k.stakingKeeper.GetDelegatorValidator(ctx, delAddr, valAddr)
 	if err != nil {
-		return nil
+		return utils.LavaFormatWarning("delegation removed hook failed", err,
+			utils.Attribute{Key: "validator_address", Value: valAddr.String()},
+			utils.Attribute{Key: "delegator_address", Value: delAddr.String()},
```
Comment (shleikes) [inline x/dualstaking/keeper/hooks.go:157]: You can use `utils.LogAttr` instead, it's shorter

File: x/dualstaking/keeper/hooks_test.go [hunk 5]
```diff
@@ -101,15 +101,16 @@ func TestReDelegateToValidator(t *testing.T) {
 func TestReDelegateToProvider(t *testing.T) {
 	ts := newTester(t)
 	ts.addValidators(1)
-	ts.addProviders(1)
+	err := ts.addProviders(1)
+	require.Nil(t, err)
```
Comment (shleikes) [inline x/dualstaking/keeper/hooks_test.go:105]:
```suggestion
	require.NoError(t, err)
```

File: x/dualstaking/keeper/hooks_test.go [hunk 6]
```diff
@@ -166,7 +167,8 @@ func TestReDelegateToProvider(t *testing.T) {
 func TestUnbondUniformProviders(t *testing.T) {
 	ts := newTester(t)
 	ts.addValidators(1)
-	ts.addProviders(5)
+	err := ts.addProviders(5)
+	require.Nil(t, err)
```
Comment (shleikes) [inline x/dualstaking/keeper/hooks_test.go:171]:
```suggestion
	require.NoError(t, err)
```

File: x/dualstaking/keeper/hooks_test.go [hunk 7]
```diff
@@ -278,7 +280,8 @@ func TestValidatorSlash(t *testing.T) {
 func TestValidatorAndProvidersSlash(t *testing.T) {
 	ts := newTester(t)
 	ts.addValidators(1)
-	ts.addProviders(5)
+	err := ts.addProviders(5)
+	require.Nil(t, err)
```
Comment (shleikes) [inline x/dualstaking/keeper/hooks_test.go:284]:
```suggestion
	require.NoError(t, err)
```

File: x/pairing/keeper/cu_tracker_test.go [hunk 8]
```diff
@@ -26,7 +26,8 @@ func TestAddingTrackedCuWithoutPay(t *testing.T) {
 	_, provider1Addr := ts.GetAccount(common.PROVIDER, 0)
 	_, provider2Addr := ts.GetAccount(common.PROVIDER, 1)
 
-	ts.TxSubscriptionBuy(client1Addr, client1Addr, "free", 1, false, false) // extend by a month so the sub won't expire
+	_, err := ts.TxSubscriptionBuy(client1Addr, client1Addr, "free", 1, false, false) // extend by a month so the sub won't expire
+	require.Nil(t, err)
```
Comment (shleikes) [inline x/pairing/keeper/cu_tracker_test.go:30]:
```suggestion
	require.NoError(t, err)
```

File: x/pairing/keeper/cu_tracker_test.go [hunk 9]
```diff
@@ -178,7 +179,8 @@ func TestTrackedCuWithQos(t *testing.T) {
 	provider1Acc, provider1 := ts.GetAccount(common.PROVIDER, 0)
 	provider2Acc, provider2 := ts.GetAccount(common.PROVIDER, 1)
 
-	ts.TxSubscriptionBuy(client, client, "free", 1, false, false) // extend by a month so the sub won't expire
+	_, err := ts.TxSubscriptionBuy(client, client, "free", 1, false, false) // extend by a month so the sub won't expire
+	require.Nil(t, err)
```
Comment (shleikes) [inline x/pairing/keeper/cu_tracker_test.go:183]:
```suggestion
	require.NoError(t, err)
```

File: x/pairing/keeper/cu_tracker_test.go [hunk 10]
```diff
@@ -381,15 +383,16 @@ func TestProviderMonthlyPayoutQuery(t *testing.T) {
 	clientAcc, client := ts.GetAccount(common.CONSUMER, 0)
 	providerAcct, provider := ts.GetAccount(common.PROVIDER, 0)
 
-	ts.TxSubscriptionBuy(client, client, "free", 1, false, false) // extend by a month so the sub won't expire
+	_, err := ts.TxSubscriptionBuy(client, client, "free", 1, false, false) // extend by a month so the sub won't expire
+	require.Nil(t, err)
```
Comment (shleikes) [inline x/pairing/keeper/cu_tracker_test.go:387]:
```suggestion
	require.NoError(t, err)
```

File: x/pairing/keeper/cu_tracker_test.go [hunk 11]
```diff
@@ -657,7 +660,8 @@ func TestTrackedCuDeletion(t *testing.T) {
 	clientAcc, client := ts.GetAccount(common.CONSUMER, 0)
 	_, provider := ts.GetAccount(common.PROVIDER, 0)
 
-	ts.TxSubscriptionBuy(client, client, "free", 1, false, false) // extend by a month so the sub won't expire
+	_, err := ts.TxSubscriptionBuy(client, client, "free", 1, false, false) // extend by a month so the sub won't expire
+	require.Nil(t, err)
```
Comment (shleikes) [inline x/pairing/keeper/cu_tracker_test.go:664]:
```suggestion
	require.NoError(t, err)
```

File: x/pairing/keeper/delegator_rewards_test.go [hunk 12]
```diff
@@ -72,7 +72,8 @@ func TestProviderDelegatorsRewards(t *testing.T) {
 			_, delegator1 := ts.GetAccount(common.CONSUMER, 1)
 			_, delegator2 := ts.GetAccount(common.CONSUMER, 2)
 
-			ts.TxSubscriptionBuy(client, client, "free", 1, false, false) // extend by a month so the sub won't expire
+			_, err := ts.TxSubscriptionBuy(client, client, "free", 1, false, false) // extend by a month so the sub won't expire
+			require.Nil(t, err)
```
Comment (shleikes) [inline x/pairing/keeper/delegator_rewards_test.go:76]:
```suggestion
			require.NoError(t, err)
```

File: x/pairing/keeper/delegator_rewards_test.go [hunk 13]
```diff
@@ -203,12 +204,13 @@ func TestDelegationLimitAffectingProviderReward(t *testing.T) {
 
 	ts.AdvanceEpoch() // to apply pairing
 
-	ts.TxSubscriptionBuy(client, client, "free", 1, false, false) // extend by a month so the sub won't expire
+	_, err := ts.TxSubscriptionBuy(client, client, "free", 1, false, false) // extend by a month so the sub won't expire
+	require.Nil(t, err)
```
Comment (shleikes) [inline x/pairing/keeper/delegator_rewards_test.go:208]:
```suggestion
	require.NoError(t, err)
```

File: x/pairing/keeper/delegator_rewards_test.go [hunk 14]
```diff
@@ -251,12 +253,13 @@ func TestProviderRewardWithCommission(t *testing.T) {
 	clientAcc, client := ts.GetAccount(common.CONSUMER, 0)
 	delegator1Acc, delegator1 := ts.GetAccount(common.CONSUMER, 1)
 
-	ts.TxSubscriptionBuy(client, client, "free", 1, false, false) // extend by a month so the sub won't expire
+	_, err := ts.TxSubscriptionBuy(client, client, "free", 1, false, false) // extend by a month so the sub won't expire
+	require.Nil(t, err)
```
Comment (shleikes) [inline x/pairing/keeper/delegator_rewards_test.go:257]:
```suggestion
	require.NoError(t, err)
```

File: x/pairing/keeper/delegator_rewards_test.go [hunk 15]
```diff
@@ -355,7 +358,8 @@ func TestQueryDelegatorRewards(t *testing.T) {
 	require.NotZero(t, currentDurationLeft)
 
 	monthsToBuy := 2
-	ts.TxSubscriptionBuy(client, client, ts.plan.Index, monthsToBuy, false, false) // extend by a month so the sub won't expire
+	_, err = ts.TxSubscriptionBuy(client, client, ts.plan.Index, monthsToBuy, false, false) // extend by a month so the sub won't expire
+	require.Nil(t, err)
```
Comment (shleikes) [inline x/pairing/keeper/delegator_rewards_test.go:362]:
```suggestion
	require.NoError(t, err)
```

File: x/pairing/keeper/msg_server_relay_payment.go [hunk 16]
```diff
@@ -76,7 +76,7 @@ func (k msgServer) RelayPayment(goCtx context.Context, msg *types.MsgRelayPaymen
 			)
 			continue
 		}
-		if relay.Epoch > ctx.BlockHeight() {
+		if relay.Epoch > ctx.BlockHeight() || relay.Epoch < 0 {
```
Comment (shleikes) [inline x/pairing/keeper/msg_server_relay_payment.go:79]: Can'y be `0` too, right? Maybe should change to `<=`?

File: x/pairing/keeper/msg_server_relay_payment_gov_test.go [hunk 17]
```diff
@@ -259,7 +259,8 @@ func TestRelayPaymentGovEpochToSaveDecrease(t *testing.T) {
 	client1Acct, client := ts.GetAccount(common.CONSUMER, 0)
 	providerAcct, providerAddr := ts.GetAccount(common.PROVIDER, 0)
 
-	ts.TxSubscriptionBuy(client, client, "free", 1, false, false) // extend by a month so the sub won't expire
+	_, err := ts.TxSubscriptionBuy(client, client, "free", 1, false, false) // extend by a month so the sub won't expire
+	require.Nil(t, err)
```
Comment (shleikes) [inline x/pairing/keeper/msg_server_relay_payment_gov_test.go:263]:
```suggestion
	require.NoError(t, err)
```

File: x/pairing/keeper/msg_server_relay_payment_test.go [hunk 18]
```diff
@@ -746,7 +746,8 @@ func TestBadgeUsedCuMapTimeout(t *testing.T) {
 	client1Acct, client1Addr := ts.GetAccount(common.CONSUMER, 0)
 	providerAcct, providerAddr := ts.GetAccount(common.PROVIDER, 0)
 
-	ts.TxSubscriptionBuy(client1Addr, client1Addr, "free", 1, false, false) // extend by a month so the sub won't expire
+	_, err := ts.TxSubscriptionBuy(client1Addr, client1Addr, "free", 1, false, false) // extend by a month so the sub won't expire
+	require.Nil(t, err)
```
Comment (shleikes) [inline x/pairing/keeper/msg_server_relay_payment_test.go:750]:
```suggestion
	require.NoError(t, err)
```

File: x/pairing/keeper/msg_server_stake_provider_test.go [hunk 19]
```diff
@@ -69,7 +69,8 @@ func TestModifyStakeProviderWithMoniker(t *testing.T) {
 
 	// modify moniker
 	moniker = "anotherExampleMoniker"
-	ts.StakeProviderExtra(providerAddr, ts.spec, testStake, nil, 0, moniker)
+	err = ts.StakeProviderExtra(providerAddr, ts.spec, testStake, nil, 0, moniker)
+	require.Nil(t, err)
```
Comment (shleikes) [inline x/pairing/keeper/msg_server_stake_provider_test.go:73]:
```suggestion
	require.NoError(t, err)
```

File: x/pairing/keeper/pairing_test.go [hunk 20]
```diff
@@ -1007,9 +1007,15 @@ func TestGeolocationPairingScores(t *testing.T) {
...
-	ts.TxSubscriptionBuy(premiumAddr, premiumAddr, premiumPlan.Index, 1, false, false)
+	ts.AdvanceMonths(1)
+	ts.AdvanceEpoch()
+
+	_, err = ts.TxSubscriptionBuy(freeAddr, freeAddr, freePlan.Index, 1, false, false)
+	require.NoError(t, err)
+	_, err = ts.TxSubscriptionBuy(basicAddr, basicAddr, basicPlan.Index, 1, false, false)
+	require.Nil(t, err)
+	_, err = ts.TxSubscriptionBuy(premiumAddr, premiumAddr, premiumPlan.Index, 1, false, false)
+	require.Nil(t, err)
```
Comment (shleikes) [inline x/pairing/keeper/pairing_test.go:1018]:
```suggestion
	require.NoError(t, err)
	_, err = ts.TxSubscriptionBuy(premiumAddr, premiumAddr, premiumPlan.Index, 1, false, false)
	require.NoError(t, err)
```

File: x/pairing/keeper/pairing_test.go [hunk 21]
```diff
@@ -1198,8 +1204,11 @@ func TestDuplicateProviders(t *testing.T) {
 	err := testkeeper.SimulatePlansAddProposal(ts.Ctx, ts.Keepers.Plans, []planstypes.Plan{basicPlan}, false)
 	require.Nil(t, err)
 
+	ts.AdvanceMonths(1)
 	ts.AdvanceEpoch()
-	ts.TxSubscriptionBuy(basicAddr, basicAddr, basicPlan.Index, 1, false, false)
+
+	_, err = ts.TxSubscriptionBuy(basicAddr, basicAddr, basicPlan.Index, 1, false, false)
+	require.Nil(t, err)
```
Comment (shleikes) [inline x/pairing/keeper/pairing_test.go:1211]:
```suggestion
	require.NoError(t, err)
```

File: x/pairing/keeper/pairing_test.go [hunk 22]
```diff
@@ -1246,8 +1255,10 @@ func TestNoRequiredGeo(t *testing.T) {
 	err := testkeeper.SimulatePlansAddProposal(ts.Ctx, ts.Keepers.Plans, []planstypes.Plan{freePlan}, false)
 	require.Nil(t, err)
 
+	ts.AdvanceMonths(1)
 	ts.AdvanceEpoch()
-	ts.TxSubscriptionBuy(freeAddr, freeAddr, freePlan.Index, 1, false, false)
+	_, err = ts.TxSubscriptionBuy(freeAddr, freeAddr, freePlan.Index, 1, false, false)
+	require.Nil(t, err)
```
Comment (shleikes) [inline x/pairing/keeper/pairing_test.go:1261]:
```suggestion
	require.NoError(t, err)
```

File: x/pairing/keeper/scores/stake_req.go [hunk 23]
```diff
@@ -17,7 +17,11 @@ func (sr *StakeReq) Init(policy planstypes.Policy) bool {
 
 // Score calculates the the provider score as the normalized stake
 func (sr *StakeReq) Score(score PairingScore) math.Uint {
-	return sdk.NewUint(score.Provider.EffectiveStake().Uint64())
+	effectiveStake := score.Provider.EffectiveStake()
+	if effectiveStake.IsNegative() || effectiveStake.IsZero() {
```
Comment (shleikes) [inline x/pairing/keeper/scores/stake_req.go:21]: I think there's a `IsPositive()` method, if I'm not wrong

File: x/rewards/keeper/rewards.go [hunk 24]
```diff
@@ -55,9 +55,9 @@ func (k Keeper) RefillRewardsPools(ctx sdk.Context, _ []byte, data []byte) {
 	// get the months left for the allocation pools
 	var monthsLeft uint64
 	if len(data) == 0 {
-		monthsLeft = uint64(types.RewardsAllocationPoolsLifetime)
+		monthsLeft = types.RewardsAllocationPoolsLifetime
 	} else {
-		monthsLeft = binary.BigEndian.Uint64(data)
+		monthsLeft = binary.BigEndian.Uint64(data) // TODO: should we check data or trust binary package?
```
Comment (shleikes) [inline x/rewards/keeper/rewards.go:60]: Are you going to resolve this TODO?

File: x/rewards/keeper/rewards.go [hunk 25]
```diff
@@ -159,5 +159,5 @@ func (k Keeper) AllocationPoolMonthsLeft(ctx sdk.Context) int64 {
 		return math.MaxInt64
 	}
 
-	return int64(binary.BigEndian.Uint64(data[0]))
+	return int64(binary.BigEndian.Uint64(data[0])) // TODO: should we check data or trust binary package?
```
Comment (shleikes) [inline x/rewards/keeper/rewards.go:162]: Are you going to resolve this TODO?

File: x/subscription/keeper/subscription_test.go [hunk 26]
```diff
@@ -365,7 +365,8 @@ func TestMonthlyRechargeCU(t *testing.T) {
 
 			// force fixation entry (by adding project key)
 			projKey := []projectstypes.ProjectKey{projectstypes.ProjectAdminKey(adm1Addr)}
-			ts.Keepers.Projects.AddKeysToProject(ts.Ctx, projectstypes.ADMIN_PROJECT_NAME, tt.developer, projKey)
+			err = ts.Keepers.Projects.AddKeysToProject(ts.Ctx, projectstypes.ADMIN_PROJECT_NAME, tt.developer, projKey)
+			require.Nil(t, err)
```
Comment (shleikes) [inline x/subscription/keeper/subscription_test.go:369]:
```suggestion
			require.NoError(t, err)
```

File: x/subscription/keeper/subscription_test.go [hunk 27]
```diff
@@ -459,9 +460,10 @@ func TestSubscriptionExpire(t *testing.T) {
 	plan := ts.Plan("free")
 
 	coins := common.NewCoins(ts.TokenDenom(), 10000)
-	ts.Keepers.BankKeeper.SetBalance(ts.Ctx, sub1Acct.Addr, coins)
+	err := ts.Keepers.BankKeeper.SetBalance(ts.Ctx, sub1Acct.Addr, coins)
+	require.Nil(t, err)
```
Comment (shleikes) [inline x/subscription/keeper/subscription_test.go:464]:
```suggestion
	require.NoError(t, err)
```

**Review by shleikes — CHANGES_REQUESTED** [review]

File: x/pairing/keeper/msg_server_relay_payment.go (hunk 16 above)
Comment (oren-lava) [reply x/pairing/keeper/msg_server_relay_payment.go:79]: Technically it can be 0, so <= is not true

**Comment by oren-lava** [comment]
> Are you sure that the command you ran run `staticcheck`? Because I ran `staticcheck` and I get more errors about unused functions, unused consts etc.

I ran it through golangci-lint as shown above. You're right, running the actual staticcheck does output more errors. These will not be handled here

File: x/subscription/keeper/subscription_test.go (hunk 26 above)
Comment (oren-lava) [reply x/subscription/keeper/subscription_test.go]: done. see https://github.com/lavanet/lava/commit/133f0e1fb5a46436cdcb891805b998fe1027074f

File: x/pairing/keeper/pairing_test.go (hunk 22 above)
Comment (oren-lava) [reply x/pairing/keeper/pairing_test.go]: done. see https://github.com/lavanet/lava/commit/133f0e1fb5a46436cdcb891805b998fe1027074f

File: x/pairing/keeper/pairing_test.go (hunk 21 above)
Comment (oren-lava) [reply x/pairing/keeper/pairing_test.go]: done. see https://github.com/lavanet/lava/commit/133f0e1fb5a46436cdcb891805b998fe1027074f

File: x/pairing/keeper/pairing_test.go (hunk 20 above)
Comment (oren-lava) [reply x/pairing/keeper/pairing_test.go]: done. see https://github.com/lavanet/lava/commit/133f0e1fb5a46436cdcb891805b998fe1027074f

File: x/pairing/keeper/msg_server_stake_provider_test.go (hunk 19 above)
Comment (oren-lava) [reply x/pairing/keeper/msg_server_stake_provider_test.go]: done. see https://github.com/lavanet/lava/commit/133f0e1fb5a46436cdcb891805b998fe1027074f

File: x/pairing/keeper/msg_server_relay_payment_test.go (hunk 18 above)
Comment (oren-lava) [reply x/pairing/keeper/msg_server_relay_payment_test.go]: done. see https://github.com/lavanet/lava/commit/133f0e1fb5a46436cdcb891805b998fe1027074f

File: x/pairing/keeper/msg_server_relay_payment_gov_test.go (hunk 17 above)
Comment (oren-lava) [reply x/pairing/keeper/msg_server_relay_payment_gov_test.go]: done. see https://github.com/lavanet/lava/commit/133f0e1fb5a46436cdcb891805b998fe1027074f

File: x/pairing/keeper/delegator_rewards_test.go (hunk 15 above)
Comment (oren-lava) [reply x/pairing/keeper/delegator_rewards_test.go]: done. see https://github.com/lavanet/lava/commit/133f0e1fb5a46436cdcb891805b998fe1027074f

File: x/pairing/keeper/delegator_rewards_test.go (hunk 14 above)
Comment (oren-lava) [reply x/pairing/keeper/delegator_rewards_test.go]: done. see https://github.com/lavanet/lava/commit/133f0e1fb5a46436cdcb891805b998fe1027074f

File: x/pairing/keeper/delegator_rewards_test.go (hunk 13 above)
Comment (oren-lava) [reply x/pairing/keeper/delegator_rewards_test.go]: done. see https://github.com/lavanet/lava/commit/133f0e1fb5a46436cdcb891805b998fe1027074f

File: x/pairing/keeper/delegator_rewards_test.go (hunk 12 above)
Comment (oren-lava) [reply x/pairing/keeper/delegator_rewards_test.go]: done. see https://github.com/lavanet/lava/commit/133f0e1fb5a46436cdcb891805b998fe1027074f

File: x/pairing/keeper/cu_tracker_test.go (hunk 11 above)
Comment (oren-lava) [reply x/pairing/keeper/cu_tracker_test.go]: done. see https://github.com/lavanet/lava/commit/133f0e1fb5a46436cdcb891805b998fe1027074f

File: x/pairing/keeper/cu_tracker_test.go (hunk 10 above)
Comment (oren-lava) [reply x/pairing/keeper/cu_tracker_test.go]: done. see https://github.com/lavanet/lava/commit/133f0e1fb5a46436cdcb891805b998fe1027074f

File: x/pairing/keeper/cu_tracker_test.go (hunk 9 above)
Comment (oren-lava) [reply x/pairing/keeper/cu_tracker_test.go]: done. see https://github.com/lavanet/lava/commit/133f0e1fb5a46436cdcb891805b998fe1027074f

File: x/pairing/keeper/cu_tracker_test.go (hunk 8 above)
Comment (oren-lava) [reply x/pairing/keeper/cu_tracker_test.go]: done. see https://github.com/lavanet/lava/commit/133f0e1fb5a46436cdcb891805b998fe1027074f

File: x/dualstaking/keeper/hooks_test.go (hunk 7 above)
Comment (oren-lava) [reply x/dualstaking/keeper/hooks_test.go]: done. see https://github.com/lavanet/lava/commit/133f0e1fb5a46436cdcb891805b998fe1027074f

File: x/dualstaking/keeper/hooks_test.go (hunk 5 above)
Comment (oren-lava) [reply x/dualstaking/keeper/hooks_test.go]: done. see https://github.com/lavanet/lava/commit/133f0e1fb5a46436cdcb891805b998fe1027074f

File: x/dualstaking/keeper/hooks.go (hunk 4 above)
Comment (oren-lava) [reply x/dualstaking/keeper/hooks.go]: done. see https://github.com/lavanet/lava/commit/133f0e1fb5a46436cdcb891805b998fe1027074f

File: x/dualstaking/keeper/helpers_test.go (hunk 3 above)
Comment (oren-lava) [reply x/dualstaking/keeper/helpers_test.go]: done. see https://github.com/lavanet/lava/commit/133f0e1fb5a46436cdcb891805b998fe1027074f

File: x/dualstaking/keeper/helpers_test.go (hunk 2 above)
Comment (oren-lava) [reply x/dualstaking/keeper/helpers_test.go]: done. see https://github.com/lavanet/lava/commit/133f0e1fb5a46436cdcb891805b998fe1027074f

File: x/pairing/keeper/scores/stake_req.go (hunk 23 above)
Comment (oren-lava) [reply x/pairing/keeper/scores/stake_req.go]: done. see https://github.com/lavanet/lava/commit/1e2fd6253b305d69b704b5f57f3f6483601ac4af

File: x/rewards/keeper/rewards.go (hunk 24 above)
Comment (oren-lava) [reply x/rewards/keeper/rewards.go:60]: I checked it and forgot to delete. fixed https://github.com/lavanet/lava/commit/47cf05ecbab2e476059ef74e97c8c31f867fc287

File: x/rewards/keeper/rewards.go (hunk 25 above)
Comment (oren-lava) [reply x/rewards/keeper/rewards.go]: I checked it and forgot to delete. fixed https://github.com/lavanet/lava/commit/47cf05ecbab2e476059ef74e97c8c31f867fc287

File: x/subscription/keeper/adjustment.go [hunk 28]
```diff
@@ -166,7 +166,8 @@ func (k Keeper) GetAdjustmentFactorProvider(ctx sdk.Context, adjustments []types
 	}
 	providers := []string{}
 	providerUsage := map[string]usage{}
-	for _, adjustment := range adjustments {
+	for _, adj := range adjustments {
```
Comment (Yaroms) [inline x/subscription/keeper/adjustment.go:169]: is this correct?
when using range isnt adj a copy?
i dont think it is the actual object from the slice

Comment (oren-lava) [reply x/subscription/keeper/adjustment.go]: reverted. see https://github.com/lavanet/lava/commit/cadc944ac3ff3648cf98a7751411e897bd5f6cfc

**Review by Yaroms — APPROVED** [review]

**Pull request merged**